library-aux/src/main/scala/scala/runtime/Statics.scala

```diff
@@ -34,56 +34,35 @@ object Statics {
   }
 
   def longHash(lv: Long): Int = {
-    lv.toInt
-    /*
-    val iv = lv.toInt | 0
+    val iv = lv.toInt
     if (iv == lv) iv
-    else (lv ^ (lv >>> 32)).toInt | 0
-    */
+    else iv ^ (lv >>> 32).toInt
   }
 
   def doubleHash(dv: Double): Int = {
-    dv.toInt
-    /*
-    val iv = dv.toInt | 0
-    if (iv == dv)
-      return iv
-
-    val fv = dv.toFloat
-    if (fv == dv)
-      return java.lang.Float.floatToIntBits(fv)
-
-    val lv = dv.toLong
-    if (lv == dv)
-      return lv.toInt | 0
-
-    val lv2 == java.lang.Double.doubleToLongBits(dv)
-    return (lv2 ^ (lv2 >>> 32)).toInt | 0
-    */
+    val iv = dv.toInt
+    if (iv == dv) {
+      iv
+    } else {
+      // don't test the case dv.toFloat == dv
+      val lv = dv.toLong
+      if (lv == dv)
+        lv.hashCode()
```

Comment:
What I implemented here is supposed to be consistent with [our `BoxesRunTime` implementation](https://github.com/scala-js/scala-js/blob/5c622745f20841b91d37aac567eec886a722e976/scalalib/overrides/scala/runtime/BoxesRunTime.scala#L90-L104), which, itself, is supposed to be consistent with [the JVM one](https://github.com/scala/scala/blob/6090f53ee9ab2677f732ea5cc7144cb9684b2593/src/library/scala/runtime/BoxesRunTime.java#L233-L244).

And here, I think the JVM implementation in `Statics` is not consistent with theirs in `BoxesRunTime`.